snuba/subscriptions/executor.py

```diff
@@ -23,8 +23,9 @@ def __init__(self, dataset: Dataset, executor_pool: ThreadPoolExecutor):
         self.__executor_pool = executor_pool
 
     def execute(
-        self, task: ScheduledTask[Subscription], tick: Tick, timer: Timer
```

Comment:
I also think there is value in recording stats over for the entire batch so we can see how concurrent queries are actually executing in production and affecting each other, but that seems more like an APM concern and I haven't spent any time seriously digging into see how that will work with multithreading and spans that are not cleanly nested.